### ISSUE TITLE
HyperOpt doesnt seem to perform any optimization of the hyperparameters

### ISSUE THREAD

**Issue by aloofhawk** [ray-project/ray#19943]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Tune

### What happened + What you expected to happen

When i run this code. It fires up two samples with some initial values but no optimization of those values with hyperopt happens. At the end of the run it just returns one of the two values with the lower scores. Can you please help me with what might be going wrong. I am expecting the optimized hyperparameters the same way i would get if i used hyperopt. I really want to take advantage of rays distributed algorithms to run all the executions in parallel. 

I dont think i need to pass in the steps parameters as my function is not stochastic. But even when i dont no training iterations of hyperopt happens 

### Versions / Dependencies

ray: latest, on ubuntu 20.04

### Reproduction script

```
def img_reg(im1, im2, config):
    hype1 = config["hype1"]
    hype2 = config["hype2"]
    ###. some computation ###
    return {"loss": loss, "status": STATUS_OK}

def trainable_hyperopt(config):
     for i in range(config["iterations"]):
         score = img_reg(im1, im2, config)
         tune.report(iterations = i, loss = score)

hyperopt_search = HyperOptSearch( metric="loss/loss", mode="min", n_initial_points = 20)


space = {"grad_step": hp.loguniform("grad_step",np.log(1e-4), np.log(1e-1)),
          "gc_metric_weight": hp.normal("gc_metric_weight", .1, 1e-2),
          "update_field_variance": hp.normal("update_field_variance", 10, 2)
        }
config = {
        "num_samples": 2,
        "config": {
            "iterations": 10
        }
    }
algo = HyperOptSearch(
        space,
       
        metric="loss/loss",
        mode="min",
        )

analysis = tune.run(trainable_hyperopt, search_alg=algo, **config, metric = "loss/loss", mode = "min")

```

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!
